src/compiler/emitter.ts

```diff
@@ -2386,8 +2374,7 @@ module ts {
                             i++;
                         }
                         write("[");
-                        emitList(elements, pos, i - pos, /*multiLine*/ (node.flags & NodeFlags.MultiLine) !== 0,
-                            /*trailingComma*/ elements.hasTrailingComma);
+                        emitList(elements, pos, i - pos, multiLine, trailingComma);

```

Comment:
I think passing trailingComma is not quite correct here. Passing it means that if the entire list has a trailing comma, then each non-spread group has a trailing comma, rather than just the last one. But I don't think it makes a difference semantically. At least, I cannot think of a case where it would.